src/industry_cmd.cpp

```diff
@@ -411,16 +411,21 @@ static Foundation GetFoundation_Industry(TileIndex tile, Slope tileh)
 static void AddAcceptedCargo_Industry(TileIndex tile, CargoArray &acceptance, CargoTypes *always_accepted)
 {
 	IndustryGfx gfx = GetIndustryGfx(tile);
-	const IndustryTileSpec *itspec = GetIndustryTileSpec(gfx);
 	const Industry *ind = Industry::GetByTile(tile);
+	const IndustryTileSpec *itspec = GetIndustryTileSpec(gfx);
+	const IndustrySpec *indspec = GetIndustrySpec(ind->type);
+
+	bool is_vanilla_industry = indspec->grf_prop.grffile == nullptr;
+	bool is_vanilla_industrytile = itspec->grf_prop.grffile == nullptr;
 
 	/* Starting point for acceptance */
 	CargoID accepts_cargo[lengthof(itspec->accepts_cargo)];
 	int8 cargo_acceptance[lengthof(itspec->acceptance)];
 	MemCpyT(accepts_cargo, itspec->accepts_cargo, lengthof(accepts_cargo));
 	MemCpyT(cargo_acceptance, itspec->acceptance, lengthof(cargo_acceptance));
 
-	if (itspec->special_flags & INDTILE_SPECIAL_ACCEPTS_ALL_CARGO) {
+	/* Non-NewGRF (vanilla) industry tiles accept all cargoes of that industry. */
+	if ((itspec->special_flags & INDTILE_SPECIAL_ACCEPTS_ALL_CARGO) || (is_vanilla_industry && is_vanilla_industrytile)) {
```

Comment:
a single variable would do, surely? :p No need for it to be at the top of the function either, put it near usage. Not writing C89...